src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const splittedDate = date.split(separator);
+  const lengthSplitedDate = splittedDate.length;
+  const storeDateObj = {};
+
+  for (let i = 0; i < lengthSplitedDate; i++) {
+    const keyFromFormat = fromFormat[i];
+    const valueFromSplitteDate = splittedDate[i];
+
+    if (keyFromFormat === 'YYYY') {
+      storeDateObj['YY'] = valueFromSplitteDate.slice(2);
+    }
+
+    if (keyFromFormat === 'YY') {
+      if (+valueFromSplitteDate < 30) {
+        storeDateObj['YYYY'] = '20' + valueFromSplitteDate;
+      } else {
+        storeDateObj['YYYY'] = '19' + valueFromSplitteDate;
+      }
```

Comment:
basically you do almost the same operations here, try to use ternary operator or create a variable for `20` or `19` and then add it to your obj, follow DRY

Reply:
okay. fixed with ternary